### PR TITLE
Lists source of all installed and mirrored packages, resolves #711

### LINKED ISSUE #711: Document and consolidate software installation sources
From [Jim Salter](https://arstechnica.com/gadgets/2017/05/how-to-build-your-own-vpn-if-youre-rightfully-wary-of-commercial-options/?comments=1&post=33391773):
> There's also a mishmash of repositories used for installing the software - a little from your distro, a lot from pip. Are you automatically updating the stuff you installed with pip? If so, how, and how frequently?

We should try and form a comprehensive picture of where the software Streisand provisions comes from. As pointed out, currently there is a mix of PPAs, source compilation, and pip. Ideally we could reduce the number of distinct sources and ensure that the ones that are easiest to automatically update are preferred.

### PR DESCRIPTION
**Pull request by nickolasclarke**
Can I get a sanity check on this approach?

### CODE REVIEW

File: documentation/SOURCES.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  - [`stunnel4`](https://github.com/jlund/streisand/blob/master/playbooks/roles/stunnel/tasks/main.yml)
+  - [`tinyproxy`](https://github.com/jlund/streisand/blob/master/playbooks/roles/tinyproxy/tasks/main.yml)
+  - [`ufw`](https://github.com/jlund/streisand/blob/master/playbooks/roles/ufw/tasks/main.yml)
+- from 3rd party repositories:
+  - [`nginx` (from nginx.org)](https://github.com/jlund/streisand/blob/master/playbooks/roles/nginx/tasks/main.yml)
+  - [`obbfs4proxy` (from deb.torproject.org)](https://github.com/jlund/streisand/blob/master/playbooks/roles/tor-bridge/tasks/main.yml)
+  - [Wireguard Packages (from wireguard PPA)](https://github.com/jlund/streisand/blob/master/playbooks/roles/wireguard/tasks/install.yml)
+
+## Source
+- [Libreswan (from official cloudfront CDN mirror)](https://github.com/jlund/streisand/blob/master/playbooks/roles/l2tp-ipsec/vars/main.yml)
```
Comment (cpu) [inline documentation/SOURCES.md]: I don't think this or the OpenConnect mirror are official in the sense that upstream runs them. They're mirrored by @jlund ad-hoc. I would maybe just leave out the `()` and say it's from source for now.

**Comment by nickolasclarke** [comment]
humm, it seems that squashing those commits removed them from here and thus your comments. Not a big deal for something like this, but what is the correct workflow to preserve changes in a PR but not pollute the master branch's commit history? @cpu

**Comment by nickolasclarke** [comment]
oh ,nvm, your review was simply auto-collapsed once I pushed the new, squashed commit. @cpu

**Comment by cpu** [comment]
@nickolasclarke I don't think we've codified a preferred practice for this repo. My own preference is to follow what I do at work: encourage review feedback to be added to open PRs as new commits on top without rebasing and then once approved & ready, squash-merging into one commit to master. This has the advantage of making review easy to follow while still keeping master clean.

**Comment by nickolasclarke** [comment]
sounds good to me! thanks

**Pull request merged**